### PR TITLE
Added an API for backup.

### PR DESCRIPTION
**Pull request by monadbobo**
This is an unfinished PR and there are still some interfaces to be implemented.

### CODE REVIEW

File: src/common/interface/meta.thrift [hunk 1]
```diff
@@ -1040,4 +1053,5 @@ service MetaService {
     ListGroupsResp listGroups(1: ListGroupsReq req);
 
     CreateBackupResp createBackup(1: CreateBackupReq req);
+    ExecResp      restoreMeta(1: RestoreMetaReq req);
```
Comment (darionyaphet) [inline src/common/interface/meta.thrift:1109]: alignment ?

File: src/common/interface/meta.thrift [hunk 2]
```diff
@@ -999,7 +999,7 @@ struct BackupMeta {
     // space_name => SpaceBackupInfo
     1: map<common.GraphSpaceID, SpaceBackupInfo> (cpp.template = "std::unordered_map")  backup_info,
     // sst file
-    2: list<binary>                               meta_files,
+    2: list<string>                               meta_files,
```
Comment (darionyaphet) [inline src/common/interface/meta.thrift:1002]: why don't use `binary`?  seems binary could support better compatibility

File: src/common/interface/meta.thrift [hunk 3]
```diff
@@ -981,6 +989,48 @@ struct GetStatisResp {
...
+    1: map<common.GraphSpaceID, SpaceBackupInfo> (cpp.template = "std::unordered_map")  backup_info,
+    // sst file
+    2: list<string>                               meta_files,
+    // backup
+    3: string                                     backup_name,
+}
+
+struct CreateBackupReq {
+    // null means all spaces
+    1: optional list<binary>  space_name,
```
Comment (bright-starry-sky) [inline src/common/interface/meta.thrift:1015]: `spaces`  is better？

File: src/common/interface/meta.thrift (hunk 2 above)
Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:1008]: Suggest using binary

**Review by bright-starry-sky — COMMENTED** [review]
Where is restore in MetaClient？

File: src/common/interface/storage.thrift [hunk 4]
```diff
@@ -818,6 +821,21 @@ struct RebuildIndexRequest {
...
+struct PartitionInfoResp {
+    1: required ResponseCommon          result,
+    2: binary                           backup_name,
+    3: common.PartitionBackupInfo       partition_info,
+}
+
+struct PartitionInfoRequest {
+    1: common.GraphSpaceID                  space_id,
+    2: binary                               backup_name,
+}
```
Comment (panda-sheep) [inline src/common/interface/storage.thrift:838]: where this code is used?

**Review by panda-sheep — COMMENTED** [review]
Good job！

File: src/common/interface/storage.thrift (hunk 4 above)
Comment (monadbobo) [reply src/common/interface/storage.thrift:838]: This information is intended for future incremental BRs and has not been implemented yet.

**Comment by monadbobo** [comment]
> Where is restore in MetaClient？

The current implementation of BR is a direct call from Go to the Meta interface, so there is no need to implement the corresponding interface in the metaclient.

**Review by panda-sheep — APPROVED** [review]
LGTM

File: src/common/interface/meta.thrift [hunk 5]
```diff
@@ -76,6 +76,14 @@ enum ErrorCode {
     E_BALANCER_FAILURE       = -58,
     E_JOB_NOT_FINISHED       = -59,
 
+    // Backup Failure
+    E_BACKUP_FAILURE = -60,
+    E_BACKUP_BUILDING_INDEX = -61,
+    E_BACKUP_SPACE_NOT_FOUND = -62,
+
+    // RESTORE Failure
+    E_RESTORE_FAILURE = -70,
```
Comment (darionyaphet) [inline src/common/interface/meta.thrift:85]: `Backup ` and `RESTORE ` should use the same segment ?

**Review by darionyaphet — APPROVED** [review]

**Pull request merged**